net-print/cnijfilter/cnijfilter-3.80.ebuild

```diff
@@ -0,0 +1,104 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit autotools
+
+PS="${PN}-source-${PV}-1"
+
+DESCRIPTION="CUPS drivers for some Canon inkjet printers (common part)"
+HOMEPAGE="http://support-sg.canon-asia.com/contents/SG/EN/0100469302.html"
+SRC_URI="http://gdlp01.c-wss.com/gds/3/0100004693/01/${PS}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+net +servicetools"
+
+RDEPEND="
+	>=media-libs/libpng-1.5:0=
+	>=media-libs/tiff-3.4:0=
+	>=net-print/cups-1.4
+	servicetools? (
+		>=dev-libs/libxml2-2.7.3-r2
+		>=x11-libs/gtk+-2.6:2
+	)
+"
+DEPEND="${DEPEND}
+	sys-devel/gettext
+"
+
+S="${WORKDIR}/${PS}"
+
+# Patches applied by the default src_prepare() since EAPI=6
+
+PATCHES=( \
+	"${FILESDIR}/cnijfilter-3.70-png.patch" \
+	"${FILESDIR}/cnijfilter-3.70-ppd.patch" \
+	"${FILESDIR}/cnijfilter-3.70-ppd2.patch" \
+	"${FILESDIR}/cnijfilter-3.70-libexec-cups.patch" \
+	"${FILESDIR}/cnijfilter-3.70-libexec-backend.patch" \
+	"${FILESDIR}/cnijfilter-3.80-cups1.6.patch" \
+	)
+
+_dir_build() {
+	local dirs=$1
+	local command=$2
+
+	local d
+
+	for d in ${dirs}; do
+		echo ">>> Working in: ${d}"
+		pushd ${d} >/dev/null
+		${command}
+		popd >/dev/null
+	done
+}
+
+pkg_setup() {
+	[[ -z ${LINGUAS} ]] && LINGUAS="en"
+
+	DIRS="libs pstocanonij backend"
+	use net && DIRS+=" backendnet"
+	#use servicetools && DIRS+=" cngpij cngpijmon/cnijnpr"
+	use servicetools && DIRS+=" cngpij"
+}
+
+src_configure() {
+	_dir_build "${DIRS}" "eautoreconf"
```

Comment:
Inline that loop from `_dir_build`, and don't print that `>>>` thing pretending to come from Portage. Also, `eautoreconf` belogs in `src_prepare`.

Alternatively, you could create a simple `configure.ac`+`Makefile.am` using [AC_CONFIG_SUBDIRS + SUBDIRS](https://www.gnu.org/software/automake/manual/html_node/Subpackages.html) that would let you build it like a regular package with autotools automatically recurring into subdirectories.

Reply:
Fixed